node/subsystem-util/src/nesting_sender.rs

```diff
@@ -0,0 +1,148 @@
+// Copyright 2022 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Hierarchical messages.
+//!
+//! Writing concurrent and even multithreaded by default is inconvenient and slow: No references
+//! hence lots of needless cloning and data duplication, locks, mutexes, ... We should reach
+//! for concurrency and parallelism when there is an actual need, not just because we can and it is
+//! reasonably safe in Rust.
+//!
+//! I very much agree with many points in this blog post for example:
+//!
+//! https://maciej.codes/2022-06-09-local-async.html
+//!
+//! Another very good post by Pierre (Tomaka):
+//!
+//! https://tomaka.medium.com/a-look-back-at-asynchronous-rust-d54d63934a1c
```

Comment:
nit: I noticed that these don't get turned into links:

```suggestion
//! <https://maciej.codes/2022-06-09-local-async.html>
//!
//! Another very good post by Pierre (Tomaka):
//!
//! <https://tomaka.medium.com/a-look-back-at-asynchronous-rust-d54d63934a1c>
```